src/main/kotlin/com/exactpro/th2/conn/dirty/tcp/core/Microservice.kt

```diff
@@ -459,5 +490,29 @@ class Microservice(
         private const val INPUT_QUEUE_ATTRIBUTE = "send"
         private const val RAW_QUEUE_ATTRIBUTE = "raw"
         private const val TRANSPORT_QUEUE_ATTRIBUTE = "transport-group"
+
+        fun createRootEventID(
+            bookName: String,
+            boxName: String,
+            eventRouter: MessageRouter<EventBatch>
+        ): EventID {
+            try {
+                val customBookRoot = Event
+                    .start()
+                    .endTimestamp()
+                    .name("$boxName with non-default book ${bookName}: ${Instant.now()}")
+                    .description("Root event")
+                    .status(Event.Status.PASSED)
+                    .toProto(bookName, boxName)
+                try {
+                    eventRouter.sendAll(EventBatch.newBuilder().apply { addEvents(customBookRoot) }.build())
+                } catch (e: IOException) {
+                    throw IllegalStateException("Can not send root event with custom book.", e);
+                }
+                return customBookRoot.id
+            } catch (e: JsonProcessingException) {
```

Comment:
suggestion: I would avoid nesting of try-catch blocks when we don't need that. You can replace it with something like this:

```kotlin
val customBookRoot =
  try {
    Event
      .start()
      .endTimestamp()
      .toProto()
  } catch (ex: IOException) {
    throw IllegalStateException()
  }
try {
    eventRouter.sendAll(EventBatch.newBuilder().apply { addEvents(customBookRoot) }.build())
} catch (e: IOException) {
    throw IllegalStateException("Can not send root event with custom book.", e);
}

```